### PR TITLE
🐛 Ignore file zz_generated.deepcopy.go in default scaffold for v3 test coverage

### PR DESCRIPTION
**Pull request by NikhilSharmaWe**
### Description
File [`zz_generated.deepcopy.go`](https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v3/api/v1/zz_generated.deepcopy.go) is autogenerated. We should ignore this file from the test coverage for solving the issue. Read [this](https://github.com/kubernetes-sigs/kubebuilder/issues/2645#issue-1218751619) and [follow-up](https://github.com/kubernetes-sigs/kubebuilder/issues/2645#issuecomment-1113963723) comment for more info.

### Motivation
Closes: https://github.com/kubernetes-sigs/kubebuilder/issues/2645

### CODE REVIEW

File: pkg/plugins/golang/v3/scaffolds/internal/templates/makefile.go
Comment (NikhilSharmaWe) [inline pkg/plugins/golang/v3/scaffolds/internal/templates/makefile.go:127]: @camilamacedo86 Could you please take a look. 
Is this approach used reasonable?

Comment (NikhilSharmaWe) [inline pkg/plugins/golang/v3/scaffolds/internal/templates/makefile.go:127]: @camilamacedo86 Could you please take a look.
Does removing the file from the cover file solved the issue here?
Is this approach used reasonable? Please inform.

Comment (Sher-Chowdhury) [reply pkg/plugins/golang/v3/scaffolds/internal/templates/makefile.go:127]: I have a few thoughts on this:

- why is the `rm` prefixed with "- "? I've never seen that before. Is that some kind of makefile syntax? If so, then what does it do?
- I think the the `tool cover` line needs to be prefixed with `go `
- I think another line can be added at the end to delete the tmp file, i.e. `rm cover.out.tmp`
- In general, not sure if this is the right approach. It feels more like a workaround, i.e. it still won't fix the output displayed in the `make test` output

**Review by Sher-Chowdhury — CHANGES_REQUESTED** [review]

**Comment by NikhilSharmaWe** [comment]
@Sher-Chowdhury @camilamacedo86 
When I change the comments on top of [zz_generated.deepcopy.go](https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v3/api/v1/zz_generated.deepcopy.go) to:
```
//go:build (!ignore_autogenerated && !skip)
// +build !ignore_autogenerated
// +build !skip
```

and also add -tags=skip in go test command in the makefile.

But when I run `make generate`, the changes made in the comments get lost (it changes to what it was before), why is it happening

**Comment by Sher-Chowdhury** [comment]
@NikhilSharmaWe I think that's because `make test` also runs `make generate` as one of the initial steps before running the actual tests - 

https://github.com/kubernetes-sigs/kubebuilder/blob/cb03a0cddc8ae511b69c32f362da2122c1aef3dc/pkg/plugins/golang/v3/scaffolds/internal/templates/makefile.go#L118

And `make generate` ends up recreating the [zz_generated.deepcopy.go](https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v3/api/v1/zz_generated.deepcopy.go) file from scratch again. 

So I think to test out your changes, you can just delete the "generate" string from your makefile.

File: pkg/plugins/golang/v3/scaffolds/internal/templates/makefile.go
Comment (camilamacedo86) [inline pkg/plugins/golang/v3/scaffolds/internal/templates/makefile.go:121]: The coverage today is done here: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v3/Makefile#L58-L60

So, if we create a new target only for that we need to update the make test target.
I'd prefer we change the make test target for ignoring the file

Also, I do not think that we can address this need by doing the cat/grep/rm.
We need to look for how we properly ignore files with go test for the option -coverprofile cover.out instead.
What are the available options provided by go test to do it?
Did we check that?

/hold

**Review by camilamacedo86 — CHANGES_REQUESTED** [review]
Thank you, @NikhilSharmaWe  for your contribution 🥇 
I made a few comments. I think it still needing some changes before we get this one merged

**Comment by camilamacedo86** [comment]
@NikhilSharmaWe @Sher-Chowdhury I put my POV in : https://github.com/kubernetes-sigs/kubebuilder/issues/2645#issuecomment-1126670250

So, I would suggest we close this one as defered. WDYT?
Is that make sense? Could we agree?

**Comment by camilamacedo86** [comment]
Closing this one based on the comments and discussions made on the issue.
However, please feel free to reach out if you do not agree with it.

**Pull request closed without merging**
